crates/ra_hir/src/db.rs

```diff
@@ -213,18 +213,11 @@ pub trait HirDatabase: DefDatabase + AstDatabase {
     #[salsa::invoke(crate::ty::traits::chalk::impl_datum_query)]
     fn impl_datum(&self, krate: Crate, impl_id: chalk_ir::ImplId) -> Arc<chalk_rust_ir::ImplDatum>;
 
-    #[salsa::invoke(crate::ty::traits::implements_query)]
-    fn implements(
+    #[salsa::invoke(crate::ty::traits::solve_query)]
+    fn solve(
```

Comment:
Micro nit about naming: perhaps this should be something like `chalk_solve` or `trait_solve`? 

Solve is fine within chalk itself, but, in larger context, it is a bit too ambiguous.